### PR TITLE
fix: Bulk Update Owner Removes Some Recipe Data

### PR DESCRIPTION
**Pull request by michael-genson**
## What type of PR is this?

_(REQUIRED)_

- bug

## What this PR does / why we need it:

_(REQUIRED)_

For more insight into why this happened, see: https://github.com/mealie-recipes/mealie/pull/2168/files

## Which issue(s) this PR fixes:

_(REQUIRED)_

N/A, reported on Discord

## Testing

_(fill-in or delete this section)_

Manually verified data is not lost

### CODE REVIEW

File: tests/integration_tests/user_recipe_tests/test_recipe_crud.py
Comment (boc-the-git) [inline tests/integration_tests/user_recipe_tests/test_recipe_crud.py:1]: Seems like we expect a different end result from patch, compared to put. Could be good to add validation of that difference in the test?

**Review by boc-the-git — APPROVED** [review]
Can you summarise the problem in a couple sentences?

Why is patch working for you in a way out didn't? Patch updates only the fields changed, whereas put is replacing the whole object?

I added a note suggesting adding logic into the test confirming those differences could be good.

**Comment by michael-genson** [comment]
Basically on the frontend we only have a `RecipeSummary` which doesn't have instructions, assets, etc. When calling update the backend assumes this is a `Recipe`, and defaults to empty instructions, assets, etc.

The `patch` method merges the old data with the new data before committing to the database

File: tests/integration_tests/user_recipe_tests/test_recipe_crud.py
Comment (michael-genson) [reply tests/integration_tests/user_recipe_tests/test_recipe_crud.py:1]: Unfortunately this is a frontend issue, not backend, so I can't really write a test for it. The frontend is sending a valid payload to the backend, which is accepted. It's just missing data

**Pull request merged**